components/list/list-item.js

```diff
@@ -0,0 +1,183 @@
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import ResizeObserver from 'resize-observer-polyfill';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+
+const ro = new ResizeObserver(entries => {
+	entries.forEach(entry => entry.target.resizedCallback(entry.contentRect));
+});
+
+class ListItem extends RtlMixin(LitElement) {
+	static get properties() {
+		return {
+			_breakpoint: { type: Number }
+		};
+	}
+
+	static get styles() {
+		const layout = css`
+			.d2l-list-item-flex {
+				display: flex;
+			}
+			.d2l-list-item-content {
+				position: relative;
+				width: 100%;
+			}
+			.d2l-list-item-content-flex {
+				display: flex;
+				flex-grow: 1;
+				justify-content: stretch;
+				margin: 18px 0;
+				padding: var(--d2l-list-item-content-padding, 0);
+			}
+			::slotted([slot|="illustration"]){
+				align-self: flex-start;
+				display: flex;
+				flex-grow: 0;
+				flex-shrink: 0;
+				margin-right: 0.9rem;
+				max-height: 52px;
+				max-width: 90px;
+				overflow:hidden;
+			}
+			:host([dir="rtl"]) ::slotted([slot|="illustration"]) {
+				margin-left: 0.9rem;
+				margin-right: 0rem;
+			}
+			::slotted([slot="illustration-outer"]) {
+				margin-bottom: 18px;
+				margin-top: 18px;
+			}
+			::slotted([slot="actions"]) {
+				align-self: flex-start;
+				display: flex;
+				flex-grow: 0;
+			}
+			.d2l-list-item-main {
+				width: 100%;
+			}
+		`;
+		const mainContent = css`
+			::slotted(.d2l-list-item-text) {
+				margin: 0;
+				max-height: 2.4rem;
+				overflow: hidden;
+			}
+			::slotted(.d2l-list-item-text-secondary-responsive),
+			::slotted(.d2l-list-item-text-secondary) {
+				margin: 0;
+				margin-top: 0.3rem;
+				overflow: hidden;
+			}
+			::slotted(.d2l-list-item-text-secondary-responsive) {
+				display: none;
```

Comment:
See my other comment about hardcoded responsive breakpoints. I don't think this `secondary-responsive` class should exist at all -- hiding/showing things should be up to the consumer.

Reply:
This is actually up to the consumer. Since the user has to add this class.

Reply:
Right, but they don't get to choose the breakpoint at which it gets hidden.

Reply:
I am going to try something that might make this experience be good for both ends.

Reply:
So I've updated the breakpoints so users can define them if they want to. It isn't the best way to define them since they have to do it for each `d2l-list-item` 

I did a rebase so here is the commit: https://github.com/BrightspaceUI/core/pull/180/commits/3583f3e66a90219d5a06f7dd419092ba77f3fdca